### ISSUE TITLE
HuggingChat

### ISSUE THREAD

**Issue by irthomasthomas** [irthomasthomas/undecidability#588]
- [ ] [HuggingChat](https://huggingface.co/chat/privacy)

# HuggingChat

**Privacy**

Last updated: October 4, 2023

Users of HuggingChat are authenticated through their HF user account.

By default, your conversations may be shared with the respective models' authors to improve their training data and model over time. Model authors are the custodians of the data collected by their model, even if it's hosted on our platform.

If you disable data sharing in your settings, your conversations will not be used for any downstream usage (including for research or model training purposes), and they will only be stored to let you access past conversations. You can click on the Delete icon to delete any past conversation at any moment.

🗓 Please also consult huggingface.co's main privacy policy at [https://huggingface.co/privacy](https://huggingface.co/privacy). To exercise any of your legal privacy rights, please send an email to privacy@huggingface.co.

**About available LLMs**

The goal of this app is to showcase that it is now possible to build an open source alternative to ChatGPT. 💪

For now (October 2023), it's running:

- Llama 2 70B
- CodeLlama 35B
- Falcon 180B
- Mistral 7B

**Technical details**

This app is running in a Space, which entails that the code for this UI is publicly visible inside the Space repo.

Further development takes place on the [huggingface/chat-ui GitHub repo](https://github.com/huggingface/chat-ui).

The inference backend is running the optimized text-generation-inference on HuggingFace's Inference API infrastructure.

It is therefore possible to deploy a copy of this app to a Space and customize it (swap model, add some UI elements, or store user messages according to your own Terms and conditions). You can also 1-click deploy your own instance using the Chat UI Spaces Docker template.

We welcome any feedback on this app: please participate in the public discussion at [https://huggingface.co/spaces/huggingchat/chat-ui/discussions](https://huggingface.co/spaces/huggingchat/chat-ui/discussions)

[More information](https://huggingface.co/chat/privacy)

####

**Comment by irthomasthomas** [comment]
### Related issues

### #418: openchat/openchat-3.5-1210 · Hugging Face
<details><summary>### Details</summary>Similarity score: 0.9
- [ ] [openchat/openchat-3.5-1210 · Hugging Face](https://huggingface.co/openchat/openchat-3.5-1210#conversation-templates)

#### Using the OpenChat Model

We highly recommend installing the OpenChat package and using the OpenChat OpenAI-compatible API server for an optimal experience. The server is optimized for high-throughput deployment using vLLM and can run on a consumer GPU with 24GB RAM.

- **Installation Guide**: Follow the installation guide in our repository.
- **Serving**: Use the OpenChat OpenAI-compatible API server by running the serving command from the table below. To enable tensor parallelism, append `--tensor-parallel-size N` to the serving command.

    Model | Size | Context | Weights | Serving
    ------|------|---------|---------|--------
    OpenChat 3.5 | 1210 | 7B | 8192 | `python -m ochat.serving.openai_api_server --model openchat/openchat-3.5-1210 --engine-use-ray --worker-use-ray`

- **API Usage**: Once started, the server listens at `localhost:18888` for requests and is compatible with the OpenAI ChatCompletion API specifications. Here's an example request:

    ```bash
    curl http://localhost:18888/v1/chat/completions \
      -H "Content-Type: application/json" \
      -d '{
            "model": "openchat_3.5",
            "messages": [{"role": "user", "content": "You are a large language model named OpenChat. Write a poem to describe yourself"}]
          }'
    ```
- **Web UI**: Use the OpenChat Web UI for a user-friendly experience.

##### Online Deployment

If you want to deploy the server as an online service, use the following options:

- `--api-keys sk-KEY1 sk-KEY2 ...` to specify allowed API keys
- `--disable-log-requests --disable-log-stats --log-file openchat.log` for logging only to a file.

For security purposes, we recommend using an HTTPS gateway in front of the server.

##### Mathematical Reasoning Mode

The OpenChat model also supports mathematical reasoning mode. To use this mode, include `condition: "Math Correct"` in your request.

    ```bash
    curl http://localhost:18888/v1/chat/completions \
      -H "Content-Type: application/json" \
      -d '{
            "model": "openchat_3.5",
            "condition": "Math Correct",
            "messages": [{"role": "user", "content": "10.3 − 7988.8133 = "}]
          }'
    ```

##### Conversation Templates

We provide several pre-built conversation templates to help you get started.

- **Default Mode (GPT4 Correct)**:

    ```markdown
    GPT4 Correct User: Hello<|end_of_turn|>
    GPT4 Correct Assistant: Hi<|end_of_turn|>
    GPT4 Correct User: How are you today?<|end_of_turn|>
    GPT4 Correct Assistant:
    ```

- **Mathematical Reasoning Mode**:

    ```markdown
    Math Correct User: 10.3 − 7988.8133=<|end_of_turn|>
    Math Correct Assistant:
    ```

    **NOTE**: Remember to set `<|end_of_turn|>` as end of generation token.

- **Integrated Tokenizer**: The default (GPT4 Correct) template is also available as the integrated tokenizer.chat\_template, which can be used instead of manually specifying the template.

#### Suggested labels
#### {   "label": "chat-templates",   "description": "Pre-defined conversation structures for specific modes of interaction." }</details>


### #552: LargeWorldModel/LWM-Text-Chat-1M · Hugging Face
<details><summary>### Details</summary>Similarity score: 0.89
- [ ] [LargeWorldModel/LWM-Text-Chat-1M · Hugging Face](https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M)


# LargeWorldModel/LWM-Text-Chat-1M · Hugging Face

**DESCRIPTION:**

LWM-Text-1M-Chat Model Card

**Model details**

Model type: LWM-Text-1M-Chat is an open-source model trained from LLaMA-2 on a subset of Books3 filtered data. It is an auto-regressive language model, based on the transformer architecture.

Model date: LWM-Text-1M-Chat was trained in December 2023.

Paper or resources for more information: [https://largeworldmodel.github.io/](https://largeworldmodel.github.io/)

**URL:** [https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M](https://huggingface.co/LargeWorldModel/LWM-Text-Chat-1M)

#### Suggested labels
#### {'label-name': 'Open-source Models', 'label-description': 'Models that are publicly available and open-source for usage and exploration.', 'gh-repo': 'huggingfaceco/LargeWorldModel/LWM-Text-Chat-1M', 'confidence': 56.11}</details>


### #310: Open LLM Leaderboard - a Hugging Face Space by HuggingFaceH4
<details><summary>### Details</summary>Similarity score: 0.88
- [ ] [Open LLM Leaderboard - a Hugging Face Space by HuggingFaceH4](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard)



#### Suggested labels
#### {   "key": "AI-Chatbots",   "value": "Topics related to advanced chatbot platforms integrating multiple AI models" } {   "key": "llm-evaluation",   "value": "Evaluating Large Language Models performance and behavior through human-written evaluation sets" }</details>


### #393: llm-vscode - Visual Studio Marketplace
<details><summary>### Details</summary>Similarity score: 0.87
- [ ] [llm-vscode - Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=HuggingFace.huggingface-vscode)

#### LLM-powered Development for VSCode

`llm-vscode` is a VSCode extension for all things LLM, built on top of the `llm-ls` backend. We also have extensions for `neovim`, `jupyter`, `intellij`, and previously `huggingface-vscode`.

**Note:** When using the Inference API, you may encounter limitations. Consider subscribing to the PRO plan to avoid rate limiting on the free tier. [Hugging Face Pricing](https://huggingface.co/pricing#pro)

#### 💻 Features

- **Code Completion:** Supports "ghost-text" code completion, à la Copilot.
- **Model Selection:** Requests for code generation are made via an HTTP request. You can use the Hugging Face Inference API or your own HTTP endpoint, as long as it adheres to the API specified [here](<https://huggingface.co/docs/transformers/main_classes/model#transformers.PreTrainedModel>) or [here](<https://github.com/huggingface/transformers/blob/main/transformers/pipelines/code_generation.py>). The list of officially supported models can be found in the [config template section](<https://github.com/Awesome-Turing/llm#user-configuration>).
- **Context Window:** The prompt sent to the model will always fit within the context window, using tokenizers to determine the number of tokens.
- **Code Attribution:** Hit `Cmd+shift+a` to check if the generated code is in The Stack. This is a rapid first-pass attribution check using [stack.dataportraits.org](http://stack.dataportraits.org). We check for sequences of at least 50 characters that match a Bloom filter, which means false positives are possible. A complete second pass can be done using the dedicated Stack search tool, which is a full dataset index.

#### 🚀 Installation

Install `llm-vscode` like any other VSCode extension.

By default, this extension uses `bigcode/starcoder` & Hugging Face Inference API for inference.

#### 🔑 HF API Token

Supply your HF API token (`hf.co/settings/token`) with this command:

- Open VSCode command palette `Cmd/Ctrl+Shift+P`
- Type: `Llm: Login`

If you previously logged in with `huggingface-cli login` on your system, the extension will read the token from disk.

#### ⚙ Configuration

Check the full list of configuration settings by opening your settings page `(cmd+,)` and typing `Llm`.

#### Suggested labels
#### {   "key": "llm-vscode",   "value": "VSCode extension for LLM powered development with Hugging Face Inference API" }</details>
